bundles/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/StateUtils.java

```diff
@@ -413,4 +413,48 @@ public static Command commandByItemState(State state) throws IllegalStateExcepti
         }
         return t;
     }
+
+    /**
+     * Compute the hue state from a given item state and a device type.
+     * If the item state matches the last command. the hue state is adjusted
+     * to use the values from the last hue state change. This is done to prevent
+     * Alexa reporting device errors.
+     * 
+     * @param itemState The item state
+     * @param deviceType The device type
+     * @param lastCommand The last command
+     * @param lastHueChange The last hue state change
+     * @return A hue light state
+     */
+    public static AbstractHueState adjustedColorStateFromItemState(State itemState, @Nullable DeviceType deviceType,
+            @Nullable Command lastCommand, @Nullable HueStateChange lastHueChange) {
+
+        AbstractHueState hueState = colorStateFromItemState(itemState, deviceType);
+
+        if (lastCommand != null && lastHueChange != null) {
+            if (lastCommand instanceof HSBType) {
+                if (itemState.as(HSBType.class).equals(lastCommand)) {
+                    HueStateColorBulb c = hueState.as(HueStateColorBulb.class);
```

Comment:
Usage of `as` is tricky here. Because the `as` as used with the openHAB `State` class returns a new instance. So the contract to be assumed should be the method returns a new instance. Here it's assumed it returns the same object. But if that would change this code will break here. Because the code below will than make changes on the local variable only and not on the object returned. It would be better to do an `instanceof` check and a cast :
```suggestion
                    HueStateColorBulb c = (HueStateColorBulb) hueState;
```
Same below with `HueStateBulb`